src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+function deleteRedundantNumber(numberWithRedundantNumber: number): number {
+  return Math.floor(numberWithRedundantNumber * 100) / 100;
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxSide = Math.max(this.a, this.b, this.c);
+
+    if (
+      [this.a, this.b, this.c].some((item) => item <= 0) ||
+      maxSide >= this.a + this.b + this.c - maxSide
+    ) {
+      throw new Error('your error message');
```

Comment:
Why u don't explain the error?
Do it for all such cases!
```suggestion
      throw new Error('...............');
```